src/components/manage/Widgets/useDimensions.js

```diff
@@ -0,0 +1,51 @@
+import { useState, useCallback, useEffect } from 'react';
+
+function getDimensionObject(node) {
+  const rect = node.getBoundingClientRect();
+
+  if (rect.toJSON) {
+    return rect.toJSON();
+  } else {
+    return {
+      width: rect.width,
+      height: rect.height,
+      top: rect.top || rect.y,
+      left: rect.left || rect.x,
+      x: rect.x || rect.left,
+      y: rect.y || rect.top,
+      right: rect.right,
+      bottom: rect.bottom,
+    };
+  }
+}
+
+function useDimensions() {
```

Comment:
This is super, but doesn't appear to be used, at least in the PR

Reply:
I had it in store, I did some trials, finally went for a full library solution. But I'll add it as a helper in another PR. :)